### PR TITLE
Add .NET 4.7 sample

### PR DESCRIPTION
**Pull request by richlander**
- Added 4.7 sample.
- Adopted .NET Core CLI compatible project format for 4.x projects.
- Changed approach for directory structure for 4.x projects.
- Updated aspnetapp sample (looks like it was never working in Docker).
- Reliant on https://github.com/Microsoft/dotnet-framework-docker/pull/19.
- In support of https://github.com/dotnet/designs/issues/7.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
I like the proposed directory structure.  Thoughts on refactoring the existing samples to do the same?

File: dotnetapp-4.7/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+dotnet-framework:4.7 Sample
```
Comment (MichaelSimons) [inline dotnetapp-4.7/README.md]: Are you going to update the main readme to reference this new sample?

**Comment by rakeshsinghranchi** [comment]
cc @MichaelSimons  
@richlander  LGTM. Any plan of porting this to dockerhub branch ? I believe there will be a need to update the build definition for building & pushing tags to dockerhub too.

**Comment by rakeshsinghranchi** [comment]
cc @MichaelSimons ,
@richlander , can you respond to my pervious query? Also, who is making the necessary change to build definition to produce 4.7 tag?

**Comment by richlander** [comment]
good question @rakeshsinghranchi. I am not. I am hoping @mcy94w will do it. 

/cc @shirhatti

**Pull request merged**